sui_programmability/examples/defi/sources/subscription.move

```diff
@@ -0,0 +1,103 @@
+/// Module implementing generic Subscription with limited number of uses.
+///
+/// This specific example shows how this model can be used within the AMM
+/// context to charge developers who make use of multiple shared objects
+/// which is expected to slow down all pools used.
+///
+/// The module itself makes use of HotPotato and Witness-based interface
+/// which can be implemented by other applications based on their needs.
+module defi::dev_pass {
+    use sui::tx_context::{TxContext};
+    use sui::object::{Self, Info};
+    use sui::transfer;
+
+    /// Owned object from which SingleUses are spawn
+    struct Subscription<phantom T> has key {
```

Comment:
Wondering if we should also have a function to top-up uses.

Reply:
We should! Some basic functionality would be good: destroy empty and top up